src/OrleansProviders/Storage/HierarchicalKeyStore.cs

```diff
@@ -7,9 +7,7 @@
 
 namespace Orleans.Storage
 {
-    internal class HierarchicalKeyStore :
-        MarshalByRefObject,
-        ILocalDataStore
+    internal class HierarchicalKeyStore : ILocalDataStore
```

Comment:
If I'm not mistaken, this had MarshalByRefObject because it was being used by a test via the app domain test hook... Was the test re-written? Good, it was my intention to do it at some point, but just wanted to be sure you searched for it

Reply:
Yep, I searched for it and couldn't find it - but of course it's possible I missed it.
The reason for this change, btw, was the disparity between `MarshalByRefObject` on CoreFx and FullFx. Because OrleansProviders was being built using the FullFx bootstrap code generator, it had the FullFx fields in codegen. This then caused an NRE when trying to load initialize the serializer on CoreFx

Reply:
I'm fine removing this here if it is no longer needed, but regarding the NRE, is there something we should do there? Something like lazily loading the serializers, similar to what we do with known types? Because I'm pretty sure we just generated a serializer for this, but the user didn't have any intention of using it (and in fact, we used this for remoting to an AppDomain, completely outside of Orleans SM).